lib/logstash/inputs/s3.rb

```diff
@@ -134,7 +134,8 @@ def list_new_files
           @logger.debug('S3 Input: Object Zero Length', :key => log.key)
         elsif !sincedb.newer?(log.last_modified)
           @logger.debug('S3 Input: Object Not Modified', :key => log.key)
-        elsif log.storage_class.start_with?('GLACIER')
+        elsif log.storage_class.start_with?('GLACIER') &&!
+            (!log.object.data.restore.nil? && log.object.data.restore.start_with?('ongoing-request="false", expiry-date="'))
```

Comment:
could we somehow avoid the double `!` e.g. `&& ( log.object.data.restore.nil? || ... )`
for the second part `start_with?('ongoing-request="false", expiry-date="')` just looks weird.
isn't there a better way, anything the sdk is recommending?

Reply:
I've updated my commit to have a cleaner elsif condition as per your suggestion.
Regarding the second condition, I agree it's not ideal having to analyse the text to work out if file is restored.
This is the info in the SDK documentation that explains it.  Basically, the method just returns a string with the contents of a header describing restore information:
https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/S3/Object.html#restore-instance_method

I'm happy to make this more robust if you like, i.e. parse the expiry date with a Regex into a DateTime object and check that it's in the future, and I'm happy for better suggestions on how to obtain this info.

In any case, tests are looking much better now, outside of the master branch.  Hope this helps.

Reply:
thanks - looking better. your suggestions to match with a regexp sounds good to me,
please make sure that a potential time parsing failure error does not propagate out ...

Reply:
No problem, I've just updated my pull request with that feature.  Does it look OK at this stage?  I've just seen your message about Glacier Deep archive, and you're right, this feature may well apply to that, so I can look into adding support for it.